modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/rebalance/RebalanceUtil.java

```diff
@@ -270,7 +271,9 @@ public static CompletableFuture<?>[] triggerAllTablePartitionsRebalance(
             int finalPartId = partId;
 
             futures[partId] = tableAssignmentsFut.thenCompose(tableAssignments ->
-                    updatePendingAssignmentsKeys(
+                    // TODO https://issues.apache.org/jira/browse/IGNITE-19763 We should distinguish empty stable assignments on
+                    // TODO node recovery in case of interrupted table creation, and moving from empty assignments to non-empty.
+                    tableAssignments.isEmpty() ? nullCompletedFuture() : updatePendingAssignmentsKeys(
```

Comment:
So, looks like the double check for isEmpty in the updatePendingAssignmentsKeys argument is redundant now?

Reply:
ofc yes, fixed